perma_web/perma/views/user_management.py

```diff
@@ -687,6 +695,40 @@ def manage_single_organization_user_reactivate(request, user_id):
     return reactive_user_in_group(request, user_id, 'organization_user')
 
 
+@user_passes_test_or_403(
+    lambda user: user.is_staff or user.is_registrar_user() or user.is_organization_user
+)
+def manage_single_organization_export_user_list(
+    request: HttpRequest, org_id: int
+) -> HttpResponse | JsonResponse:
+    """Return a file listing all users belonging to an organization."""
+    org_users = (
+        LinkUser.objects.filter(organizations__id=org_id)
+        .annotate(organization_name=F('organizations__name'))
+        .values('email', 'first_name', 'last_name', 'organization_name')
+    )
+    filename_stem = f'perma-organization-{org_id}-users'
+
+    # Generate output records from query results and add organization name
+    field_names = ['email', 'first_name', 'last_name', 'organization_name']
```

Comment:
I'm curious about the choice to include `organization_name` in every row. Is it for ease of testing? Or, for a future where this code can be reused to export, e.g., all the users associated with a given registrar, across all their orgs? Some other reason I'm not thinking of?

Reply:
Good question. That is my attempt to address a request in [ENG-983](https://linear.app/harvard-lil/issue/ENG-983/create-a-csv-download-of-organization-users):

> If it's possible to add [the org name] as metadata or a header in the CSV itself that's ideal — otherwise just have that as an additional populated field.

My feeling is that having a normal-looking, valid CSV (albeit with redundant values) is better than sticking the org name in the header or somewhere else. As you noted, this structure would also lend itself to cross-registrar output or similar. Thoughts?

Reply:
👍 agreed, that sounds better to me too than some kind of fancier header. and, dealing with formatting names -> filenames sounds No Fun ™️, possibly less useful...

👍 harmlessly redundant data 🙂